build/Dockerfile

```diff
@@ -4,170 +4,301 @@
+ARG MACHINE_TOOLS_VERSION
+ARG MACHINE_IMAGE_KERNEL_VERSION
+ARG MACHINE_KERNEL_VERSION
 ARG DEBIAN_FRONTEND=noninteractive
+
+# Install ca-certificates and curl (setup).
+RUN <<EOF
+    set -e
+    apt-get update
+    apt-get install -y --no-install-recommends ca-certificates curl
+EOF
+
+# Install the cartesi-machine (cartesi/machine-emulator).
 RUN <<EOF
-set -e
-apt-get update
-apt-get install -y --no-install-recommends \
-    ca-certificates \
-    curl
-ARCH=$(dpkg --print-architecture)
-curl -fsSL https://github.com/cartesi/machine-emulator/releases/download/v${MACHINE_EMULATOR_VERSION}/cartesi-machine-v${MACHINE_EMULATOR_VERSION}_${ARCH}.deb -o ./cartesi-machine.deb
-apt-get install -y ./cartesi-machine.deb
-rm ./cartesi-machine.deb
-rm -rf /var/lib/apt/lists/*
+    set -e
+    ARCH=$(dpkg --print-architecture)
+    URL=https://github.com/cartesi/machine-emulator/releases/download
+    VERSION=v${MACHINE_EMULATOR_VERSION}
+    ARTIFACT=cartesi-machine-${VERSION}_${ARCH}.deb
+    curl -fsSL ${URL}/${VERSION}/${ARTIFACT} -o ./cartesi-machine.deb
+    apt-get install -y ./cartesi-machine.deb
+    rm ./cartesi-machine.deb
 EOF
 
-# Configure cartesi user and group.
+# Configure cartesi user and cartesi group.
 RUN <<EOF
-set -e
-addgroup --system --gid 102 cartesi
-adduser --system --uid 102 \
-    --disabled-login \
-    --gecos "cartesi user" \
-    --home /nonexistent \
-    --ingroup cartesi \
-    --no-create-home \
-    --shell /bin/false \
-    cartesi
+    set -e
+    addgroup --system --gid 102 cartesi
+    adduser --system --uid 102 \
+        --disabled-login \
+        --gecos "cartesi user" \
+        --home /nonexistent \
+        --ingroup cartesi \
+        --no-create-home \
+        --shell /bin/false \
+        cartesi
 EOF
 
-####################################################################################################
-# TARGET: rollups-node-snapshot
+# Download linux.bin (cartesi/image-kernel).
+RUN <<EOF
+    set -e
+    URL=https://github.com/cartesi/image-kernel/releases/download
+    VERSION=v${MACHINE_IMAGE_KERNEL_VERSION}
+    ARTIFACT=linux-${MACHINE_KERNEL_VERSION}-ctsi-1-${VERSION}.bin
+    curl -fsSL ${URL}/${VERSION}/${ARTIFACT} -o ./linux.bin
+    mv ./linux.bin /usr/share/cartesi-machine/images
+EOF
+
+# Download rootfs.ext2 (cartesi/machine-emulator-tools).
+RUN <<EOF
+    set -e
+    URL=https://github.com/cartesi/machine-emulator-tools/releases/download
+    VERSION=v${MACHINE_TOOLS_VERSION}
+    ARTIFACT=rootfs-tools-${VERSION}.ext2
+    curl -fsSL ${URL}/${VERSION}/${ARTIFACT} -o ./rootfs.ext2
+    mv ./rootfs.ext2 /usr/share/cartesi-machine/images
+EOF
+
+# Clean up.
+RUN <<EOF
+    rm -rf /var/lib/apt/lists/*
+EOF
+
+# =============================================================================
+# STAGE: emulator-devel
+#
+# - Install libarchive13 (setup -- required by xgenext2fs).
+# - Install libcmt.
+# - Install xgenext2fs.
+#
+# IMPORTANT: Only works with the linux/amd64 platform!
```

Comment:
Perhaps we should tell which libraries add this constraint.

Reply:
Changed to:
```
# IMPORTANT: Only works with the linux/amd64 platform!
#                         (libarchive13, required by xgenext2fs, is amd64 exclusive.)
```